configure.ac

```diff
@@ -329,6 +329,7 @@ AC_HEADER_DIRENT
 AC_HEADER_STDC
 # after _HEADER_STDC
 AC_CHECK_HEADERS([sys/param.h sys/mman.h sys/select.h locale.h memory.h getopt.h unistd.h mcheck.h limits.h sys/time.h langinfo.h])
+AM_CONDITIONAL([HAVE_MMAN], [test x$ac_cv_header_sys_mman_h = xyes])
```

Comment:
The AM define for the camlibs/Makefile.am which is set according to the sys/mman.h header detection. There are better ways how to do it, but this one is simple and worked for me.

Reply:
On my Fedora 34 system, running `../configure --with-camlibs=st2205` for a native build of only the camlib in question prints the following:

```
checking sys/mman.h usability... yes
checking sys/mman.h presence... yes
checking for sys/mman.h... yes
```

and then the `make` run fails with

```
/usr/bin/ld: cannot find -lmman
```

as there is no `libmman` on the system to link `st2205.so` against.

This definition of the `HAVE_MMAN` automake conditional may be simple, but it does *not* work for me for native Linux builds. The check should be whether linking a piece of code which calls `mmap(2)` works without adding `-lmman` to the libraries to link against, and whether linking against `-lmman` actually fixes that. And such a condtional would be more accurately named something like `MMAP_NEEDS_LIBMMAN`.

However, we should take a step back here or two to see what the `st2205` camlib actually does with `mmap(2)`: It allocates a page aligned memory buffer by opening `/dev/zero`. Does that trick even work on Windows?

The comment above the `st2205_malloc_page_aligned(int size)` function (yes, `int` not `size_t`) indicates that the page aligned buffer is for cooperating with the use of `O_DIRECT` within `usbdiskdirect` iolib. This iolib is implemented by a source file `libgphoto2_port/usbdiskdirect/linux.c` whose name suggests it has been written very much with Linux in mind, and which just defines the `O_DIRECT` macro to be `00040000` if it is undefined. What that value does on Windows needs to be tested. https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/open-wopen?view=msvc-160 indicates that `O_DIRECT` does not exist on Windows.

So I would like to ask whether it even makes sense at all to just link `st2205.so` against `libmman` in a mingw build (`--with-camlibs=standard,-st2205`), or whether the `st2205` driver requires so much porting effort that it should be just deactivated in mingw builds.

Reply:
The problem with the -lmman breaking the native build should be addressed with the latest PR. But of course I can drop the st2205 driver in the mxe (that's what I initially did).